cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/visitor/ExpandHelper.java

```diff
@@ -302,4 +316,112 @@ private void expandIncluded(
+     * @param requestedExpansionParameters the {@link IParametersAdapter} representing the parameters that
+     *                                     were originally sent to the terminology service; may be {@code null},
+     *                                     in which case this method is a no-op
+     */
+    private void validateExpansionParameters(
+            IValueSetAdapter expandedValueSet, IParametersAdapter requestedExpansionParameters) {
+
+        if (expandedValueSet == null || requestedExpansionParameters == null) {
+            return;
+        }
+
+        // 1. No expansion at all -> immediate warning
+        if (!expandedValueSet.hasExpansion()) {
+            addExpansionWarningParameter(
+                    expandedValueSet,
+                    "Expansion for ValueSet %s did not use expected expansion parameters (no expansion element)."
+                            .formatted(expandedValueSet.getUrl()));
+            return;
+        }
+
+        // 2. Build map of requested parameters we care about (system-version, valueset-version) -> expected string
+        // value
+        Map<String, String> requestedValues = new HashMap<>();
+        for (String paramName : EXPANSION_PARAMETERS_TO_VALIDATE) {
+            if (!requestedExpansionParameters.hasParameter(paramName)) {
+                continue;
+            }
+
+            var valueOpt = ParametersUtil.getNamedParameterValueAsString(
+                    fhirContext(), (IBaseParameters) requestedExpansionParameters.get(), paramName);
+
+            valueOpt.filter(StringUtils::isNotBlank).ifPresent(v -> requestedValues.put(paramName, v));
+        }
+
+        if (requestedValues.isEmpty()) {
+            // We didn’t actually request any of the parameters we care about with usable values
+            return;
+        }
+
+        // 3. For each requested parameter, see if the expansion has either:
+        //    - the same name with that value, or
+        //    - the used-* variant with that value.
+        boolean mismatchOrMissing = requestedValues.entrySet().stream().anyMatch(entry -> {
+            var requestedName = entry.getKey();
+            var expectedValue = entry.getValue();
+
+            var usedName = EXPANSION_PARAMETER_USED_NAME_OVERRIDES.get(requestedName);
+
+            boolean hasRequested = expandedValueSet.hasExpansionStringParameter(requestedName, expectedValue);
+            boolean hasUsed = usedName != null && expandedValueSet.hasExpansionStringParameter(usedName, expectedValue);
+
+            // Neither name has the expected value -> treat as missing/mismatched
+            return !(hasRequested || hasUsed);
+        });
+
+        // 4. If *any* requested parameter is missing or mismatched, add a single warning parameter.
+        if (mismatchOrMissing) {
+            addExpansionWarningParameter(
+                    expandedValueSet,
+                    "Expansion for ValueSet %s did not use expected expansion parameters."
```

Comment:
I think it would be a much more useful message if it said what expected parameters weren't used.